x/cronos/keeper/ibc.go

```diff
@@ -121,10 +122,21 @@ func (k Keeper) IbcTransferCoins(ctx sdk.Context, from, destination string, coin
 			}
 
 			err = k.ibcSendTransfer(ctx, acc, destination, ibcCoin)
+			fmt.Printf("IbcTransferCoins: %+v, %+v\n", err, coins)
 			if err != nil {
+				if ibcchanneltypes.ErrPacketTimeout.Is(err) {
```

Comment:
isnt it already done in the middleware?
https://github.com/crypto-org-chain/cronos/blob/main/x/cronos/middleware/conversion_middleware.go#L121

Reply:
maybe I misunderstand timeout case, but the ErrPacketTimeout is throw when [SendPacket](https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/keeper/relay.go#L160), which hasnt reach [L121](https://github.com/crypto-org-chain/cronos/blob/main/x/cronos/middleware/conversion_middleware.go#L121) yet

Reply:
The timeout happens when the packet is received on "crypto.org chain" or it could also be invalid.

At that time, the relayer should report an aknowledgement.Error or TimeoutPacket back to "cronos?

Reply:
oh actually I think it cannot timeout because of the zero height 
https://github.com/crypto-org-chain/cronos/blob/main/x/cronos/keeper/ibc.go#L177

Reply:
and for timeout, it seems you are right, it is checked upon the sending chain on the light client state

Reply:
Maybe you can try the case with the timestamp but not sure how to do since the timestamp is assigned immediately before the check.

Anyways, I think you don't need to mint again here, if it returns error, the whole state revert

Reply:
[timeout](https://tutorials.cosmos.network/academy/4-ibc/channels.html#timeout-case) is easier with enable timeoutHeight, sender can verify with relayer to `QueryNonReceipt` from receiver, but refund seems break for after https://github.com/crypto-org-chain/cronos/pull/600/commits/cdcfda17e7b923f5970608ba289f19e7fb338fa9, we parse trace from `data.Denom` instead of `unprefixedDenom` on timeout